portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +34,67 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+    private static final int minCommentLimit = 5;
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    Query query = new Query(Comment.entityNameParam).addSort(Comment.timeParam, SortDirection.DESCENDING);
+    
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<Comment> comments = new ArrayList<>();
+
+    int commentLimit = getCommentLimit(request);
+
+    results.asList(FetchOptions.Builder.withLimit(commentLimit)).forEach(entity -> {
+      Comment comment = commentFromEntity(entity);
+      comments.add(comment);
+    });
+
+    Gson gson = new Gson();
+    String json = gson.toJson(comments);
+
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+    try {
```

Comment:
What do we need this for?

Reply:
Removed.

Reply:
I was trying to fix a problem where on occasion, a comment would not display right after it was added until the page was refreshed. I removed the delay as it didn't help.